### PR TITLE
Remove default key from keosd

### PR DESCRIPTION
**Pull request by PaulCalabrese**
Removes the default key so the wallet plugin does not load it by default.
Add a remove_key interface to the wallet_api to allow users to remove keys from the wallet.
Add "wallet remove" command to cleos.
Add new unit tests to wallet unit tests for remove_key functionality.
Fix test breakage caused by removal of the default key.

### CODE REVIEW

File: plugins/wallet_plugin/include/eosio/wallet_plugin/wallet_manager.hpp [hunk 1]
```diff
@@ -107,6 +107,13 @@ class wallet_manager {
    /// @throws fc::exception if wallet not found or locked.
    void import_key(const std::string& name, const std::string& wif_key);
 
+   /// Removes a key from the specified wallet.
+   /// Wallet must be opened and unlocked.
+   /// @param name the name of the wallet to import into.
```
Comment (heifner) [inline plugins/wallet_plugin/include/eosio/wallet_plugin/wallet_manager.hpp]: ... to remove from

File: plugins/wallet_plugin/include/eosio/wallet_plugin/wallet_manager.hpp [hunk 2]
```diff
@@ -107,6 +107,13 @@ class wallet_manager {
...
+   /// @param key the Public Key to remove, e.g. EOS6MRyAjQq8ud7hVNYcfnVPJqcVpscN5So8BhtHuGYqET5GDW5CV
+   /// @throws fc::exception if wallet not found or locked or key is not imported.
+   void remove_key(const std::string& name, const std::string& key);
```
Comment (heifner) [inline plugins/wallet_plugin/include/eosio/wallet_plugin/wallet_manager.hpp]: Needs to require password.

File: plugins/wallet_plugin/include/eosio/wallet_plugin/wallet_manager.hpp [hunk 3]
```diff
@@ -126,7 +133,7 @@ class wallet_manager {
    std::chrono::seconds timeout = std::chrono::seconds::max(); ///< how long to wait before calling lock_all()
    mutable timepoint_t timeout_time = timepoint_t::max(); ///< when to call lock_all()
    boost::filesystem::path dir = ".";
-   std::string eosio_key = "5KQwrPbwdL6PhXujxW37FSSQZ1JiwsST4cqQzDeyXtP79zkvFD3";
+   std::string eosio_key;
```
Comment (heifner) [inline plugins/wallet_plugin/include/eosio/wallet_plugin/wallet_manager.hpp]: This can be removed as not used now.

File: programs/cleos/main.cpp [hunk 4]
```diff
@@ -2216,6 +2216,23 @@ int main( int argc, char** argv ) {
       std::cout << localized("imported private key for: ${pubkey}", ("pubkey", std::string(pubkey))) << std::endl;
    });
 
+   // remove keys from wallet
+   string wallet_rm_key_str;
+   auto removeKeyWallet = wallet->add_subcommand("remove", localized("Remove key from wallet"), false);
```
Comment (heifner) [inline programs/cleos/main.cpp]: Lets name it "remove_key"

File: plugins/wallet_plugin/include/eosio/wallet_plugin/wallet_manager.hpp (hunk 3 above)
Comment (PaulCalabrese) [reply plugins/wallet_plugin/include/eosio/wallet_plugin/wallet_manager.hpp]: You can still pass it on the command line as an option.

Comment (heifner) [reply plugins/wallet_plugin/include/eosio/wallet_plugin/wallet_manager.hpp]: Please remove that feature as well.

**Pull request merged**